Experimental/src/org/sleuthkit/autopsy/experimental/volatilityDSP/AddMemoryImageTask.java

```diff
@@ -130,8 +130,15 @@ public void run() {
     private Image addImageToCase(List<String> errorMessages) {
         progressMonitor.setProgressText(Bundle.AddMemoryImageTask_progress_add_text() + imageFilePath);
         
-        SleuthkitCase caseDatabase = Case.getCurrentCase().getSleuthkitCase();
-        caseDatabase.acquireExclusiveLock();
+        SleuthkitCase caseDatabase;
+        try {
+            caseDatabase = Case.getOpenCase().getSleuthkitCase();
+        } catch (NoCurrentCaseException ex) {
+            errorMessages.add(Bundle.AddMemoryImageTask_image_critical_error_adding() + imageFilePath + Bundle.AddMemoryImageTask_for_device() + deviceId + ":" + ex.getLocalizedMessage());            
+            return null;
+        }
+
+        caseDatabase.acquireSingleUserCaseWriteLock();
```

Comment:
It seems like this lock doesn't get released if the imageFile.exists() test a few lines down fails

Reply:
Good catch.